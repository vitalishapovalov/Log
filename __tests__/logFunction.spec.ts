import { AfterEach, BeforeEach, Describe, Test } from "@jest-decorated/core";
import { getTFunction, getTObject } from "./__mocks__/logFunction";

@Describe("log/log test")
class LogFunctionSpec {

    dummy;
    logSpy;

    @BeforeEach()
    beforeEach() {
        this.logSpy = jest.spyOn(console, "log");
    }

    @AfterEach()
    afterEach() {
        this.logSpy.mockClear();
    }

    @Test("when used on an object, should correctly log actions")
    asObject() {
        const instance = getTObject();

        this.dummy = instance.prop;
        instance.prop = "";
        this.dummy = instance.getSet;
        instance.getSet = "";
        this.dummy = instance.method();

        expect(JSON.stringify(this.logSpy.mock.calls[0])).toMatchSnapshot();
        expect(JSON.stringify(this.logSpy.mock.calls[1])).toMatchSnapshot();
        expect(JSON.stringify(this.logSpy.mock.calls[2])).toMatchSnapshot();
        expect(JSON.stringify(this.logSpy.mock.calls[3])).toMatchSnapshot();
        expect(JSON.stringify(this.logSpy.mock.calls[4])).toMatchSnapshot();
        expect(this.logSpy).toHaveBeenCalledTimes(5);
    }

    @Test("when used on a function, should correctly log apply")
    asFunction() {
        const fn = getTFunction();

        fn(100, ["alex"]);

        expect(JSON.stringify(this.logSpy.mock.calls[0])).toMatchSnapshot();
        expect(this.logSpy).toHaveBeenCalledTimes(1);
    }
}
