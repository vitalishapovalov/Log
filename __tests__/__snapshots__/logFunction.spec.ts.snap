// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`log/log test when used on a function, should correctly log apply 1`] = `"[\\"\\\\u001b[40m\\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mLogger\\\\u001b[39m         \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mTFunction\\\\u001b[39m                                      \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mCall \\\\u001b[91mfunction\\\\u001b[39m\\\\u001b[39m  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[93mTFunction\\\\u001b[39m\\\\u001b[97m(\\\\u001b[39m\\\\u001b[97marg1\\\\u001b[39m\\\\u001b[97m:\\\\u001b[39m \\\\u001b[97mNumber\\\\u001b[39m\\\\u001b[91m,\\\\u001b[39m \\\\u001b[97marg2\\\\u001b[39m\\\\u001b[97m:\\\\u001b[39m \\\\u001b[97mArray\\\\u001b[39m\\\\u001b[97m)\\\\u001b[39m\\\\u001b[97m:\\\\u001b[39m \\\\u001b[97mString\\\\u001b[39m\\\\u001b[91m;\\\\u001b[39m  \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mCall context\\\\u001b[39m   \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mundefined\\\\u001b[39m                                      \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mCall with\\\\u001b[39m      \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[97m(\\\\u001b[39m\\\\u001b[36m100\\\\u001b[39m\\\\u001b[91m,\\\\u001b[39m \\\\u001b[97m[\\\\u001b[39m\\\\u001b[32m\\\\\\"alex\\\\\\"\\\\u001b[39m\\\\u001b[97m]\\\\u001b[39m\\\\u001b[97m)\\\\u001b[39m                                \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mCall result\\\\u001b[39m    \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[32m\\\\\\"value\\\\\\"\\\\u001b[39m                                        \\\\u001b[49m\\"]"`;

exports[`log/log test when used on an object, should correctly log actions 1`] = `"[\\"\\\\u001b[40m\\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mAccess property\\\\u001b[39m  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[35mprop\\\\u001b[39m\\\\u001b[97m:\\\\u001b[39m \\\\u001b[97mString\\\\u001b[39m\\\\u001b[91m;\\\\u001b[39m  \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mAccess result\\\\u001b[39m    \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[32m\\\\\\"prop_value\\\\\\"\\\\u001b[39m   \\\\u001b[49m\\"]"`;

exports[`log/log test when used on an object, should correctly log actions 2`] = `"[\\"\\\\u001b[40m\\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mSet property\\\\u001b[39m  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[35mprop\\\\u001b[39m\\\\u001b[97m:\\\\u001b[39m \\\\u001b[97mString\\\\u001b[39m\\\\u001b[91m;\\\\u001b[39m  \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mSet \\\\u001b[32msucceed\\\\u001b[39m\\\\u001b[39m   \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[35mprop\\\\u001b[39m \\\\u001b[97m=\\\\u001b[39m \\\\u001b[32m\\\\\\"\\\\\\"\\\\u001b[39m\\\\u001b[91m;\\\\u001b[39m     \\\\u001b[49m\\"]"`;

exports[`log/log test when used on an object, should correctly log actions 3`] = `"[\\"\\\\u001b[40m\\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mAccess property\\\\u001b[39m  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[91mget\\\\u001b[39m \\\\u001b[93mgetSet\\\\u001b[39m\\\\u001b[97m(\\\\u001b[39m\\\\u001b[97m)\\\\u001b[39m\\\\u001b[97m:\\\\u001b[39m \\\\u001b[97mString\\\\u001b[39m\\\\u001b[91m;\\\\u001b[39m  \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mAccess result\\\\u001b[39m    \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[32m\\\\\\"getSet_value\\\\\\"\\\\u001b[39m         \\\\u001b[49m\\"]"`;

exports[`log/log test when used on an object, should correctly log actions 4`] = `"[\\"\\\\u001b[40m\\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mSet property\\\\u001b[39m  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[35mgetSet\\\\u001b[39m\\\\u001b[97m:\\\\u001b[39m \\\\u001b[97mString\\\\u001b[39m\\\\u001b[91m;\\\\u001b[39m  \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mSet \\\\u001b[32msucceed\\\\u001b[39m\\\\u001b[39m   \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[35mgetSet\\\\u001b[39m \\\\u001b[97m=\\\\u001b[39m \\\\u001b[32m\\\\\\"\\\\\\"\\\\u001b[39m\\\\u001b[91m;\\\\u001b[39m     \\\\u001b[49m\\"]"`;

exports[`log/log test when used on an object, should correctly log actions 5`] = `"[\\"\\\\u001b[40m\\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mCall \\\\u001b[93mmethod\\\\u001b[97m\\\\u001b[39m  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[93mmethod\\\\u001b[39m\\\\u001b[97m(\\\\u001b[39m\\\\u001b[97m)\\\\u001b[39m\\\\u001b[97m:\\\\u001b[39m \\\\u001b[97mString\\\\u001b[39m\\\\u001b[91m;\\\\u001b[39m  \\\\u001b[49m\\\\n\\\\u001b[40m  \\\\u001b[97mCall result\\\\u001b[39m  \\\\u001b[37m@\\\\u001b[39m  \\\\u001b[32m\\\\\\"method_value\\\\\\"\\\\u001b[39m     \\\\u001b[49m\\"]"`;
